### PR TITLE
fix(1994): [3] add tag/release filtering

### PR DESCRIPTION
**Pull request by wahapo**
## Context
Add the guide of tag filtering. ~It will take a little longer to create~ a PR for feature. => [1] https://github.com/screwdriver-cd/models/pull/460 and [2] https://github.com/screwdriver-cd/screwdriver/pull/2110
Please check the contents ~first~.

## References
https://github.com/screwdriver-cd/screwdriver/issues/1994

[1] https://github.com/screwdriver-cd/models/pull/460
[2] https://github.com/screwdriver-cd/screwdriver/pull/2110
[3] https://github.com/screwdriver-cd/guide/pull/386 (this PR)

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: docs/ja/user-guide/configuration/workflow.md [hunk 1]
```diff
@@ -183,6 +185,31 @@ _注意: ブランチに対するPRのワークフローは、そのブランチ
 
 [ブランチフィルターのサンプルリポジトリ](https://github.com/screwdriver-cd-test/branch-filtering-example)を参考にしてください。ブランチフィルターがプルリクエストに対してどう動作するのかを見るのなら、[プルリクエストの例](https://github.com/screwdriver-cd-test/branch-filtering-example/pull/2)を参考にしてください。
 
+## タグフィルターとリリースフィルター
+タグフィルター/リリースフィルターを使用して、特定の名称の`~tag`/`~release`イベントをトリガーにすることができます。特定の名称のタグが作成された後にパイプラインでジョブをトリガーするには、`requires: [~tag:tagName]`を使用します。特定の名称のリリースがされた後にパイプラインでジョブをトリガーするには、`requires: [~release:releaseName]`を使用します。`tagName`と`releaseName`は([JavaScript仕様の](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions))正規表現を用いて指定することもできます（例: `~release：/^feature-/`）。正規表現のフラグはサポートしていません。
+
+### 例
+以下の例では、`stable`という名称のリリースがされると、`all-tag-and-release`と `stable-release`がトリガーされます。また、`v1.0`という名称のタグが作成されると、`all-tag-and-release`と `v1-tag`がトリガーされます。`v2.0`という名称のタグが作成されると、`all-tag-and-release`がトリガーされます。
```
Comment (yoshwata) [inline docs/ja/user-guide/configuration/workflow.md:192]: [ask] 仕様についての質問になっちゃうのですが、stableのリリース時に`all-tag-and-release`と`stable-release`が両方トリガされるのって良いんだっけ・・・ってなりました。いまbranch specificのときはcommitをトリガしないと思うので、対称性がないような。。。と

Comment (wahapo) [reply docs/ja/user-guide/configuration/workflow.md:192]: ~あ、確かに。それが正しいです。~
→正しくなかった…

**Review by tkyi — APPROVED** [review]

File: docs/ja/user-guide/configuration/workflow.md (hunk 1 above)
Comment (wahapo) [reply docs/ja/user-guide/configuration/workflow.md:192]: @yoshwata Currently, both `~commit` and `~commit:master` seem to work at same time, even when using the branch filtering. So also in this case, it seems better to fire both `~tag` and `~tag:tagName` at the same time.

**Comment by wahapo** [comment]
@tkyi @jithine I think this is fine, Is this correct?
https://github.com/screwdriver-cd/guide/pull/386#discussion_r439224490

File: docs/ja/user-guide/configuration/workflow.md (hunk 1 above)
Comment (tkyi) [reply docs/ja/user-guide/configuration/workflow.md:192]: Yes, I agree the two mechanisms should be consistent. Thanks for the catch

**Review by yoshwata — APPROVED** [review]

**Review by tkyi — APPROVED** [review]

**Pull request merged**